rust/bridge/ffi/src/lib.rs

```diff
@@ -260,6 +260,21 @@ pub unsafe extern "C" fn signal_session_record_archive_current_state(
     })
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn signal_session_record_has_current_state(
+    result: *mut bool,
+    session_record: *const SessionRecord,
+) -> *mut SignalFfiError {
+    run_ffi_safe(|| {
+        if result.is_null() {
+            return Err(SignalFfiError::NullPointer);
+        }
+        let session_record = native_handle_cast::<SessionRecord>(session_record)?;
+        *result = session_record.current_session.is_some();
```

Comment:
The `current_session` isn't accessible anymore, I think you'll need to add an equivalent accessor on `SessionRecord` (or you could use `session_record.session_state().is_ok()` knowing that the only reason that would fail is because there is no current session, but that's a little subtle)